### PR TITLE
Assignment 1

### PR DESCRIPTION
**Pull request by snehaldhadge**
1. Assignment1_bash.sh : Shell Script to run the Python File
2. Assignment_1.py: Python file to do all the analysis
3. Changes in pre-commit file to use https instead of ssh

### CODE REVIEW

File: Assignment_1.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import plotly.express as px
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.linear_model import LogisticRegression
+from sklearn.metrics import accuracy_score
+from sklearn.model_selection import train_test_split
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import Normalizer
+
+
+def print_heading(title):
```
Comment (vbihare) [inline Assignment_1.py:15]: This is good, this way by calling the 'print_heading' function you can easily differentiate between different sections.

File: Assignment_1.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    # Scatter Plot of different features
+    fig = px.scatter_matrix(
+        iris_df,
+        dimensions=["Sepal_Width", "Sepal_Length", "Petal_Width", "Petal_Length"],
+        color="Species",
+    )
+    fig.show()
+
+    # Pie chart to show Population of each Species in the dataset
+    fig = px.pie(
```
Comment (vbihare) [inline Assignment_1.py:99]: Developing a pie chart here won't explain much, I would suggest trying something new like the 3d scatter plot. It is really good.

**Review by vbihare — COMMENTED** [review]
Good work there! Your code is readable and quite easy to understand, with all the comments you have provided.

File: Assignment_1.py (hunk 2 above)
Comment (snehaldhadge) [reply Assignment_1.py:99]: The idea was to visualize the proportion of data for each Species.

Comment (vbihare) [reply Assignment_1.py:99]: Okay.

File: Assignment_1.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+    col_names = [
+        "Sepal_Length",
+        "Sepal_Width",
+        "Petal_Length",
+        "Petal_Width",
+        "Species",
+    ]
+    # Load the Iris data set into Pandas DataFrame
+    iris_df = pd.read_csv("Iris.data", names=col_names)
```
Comment (dafrenchyman) [inline Assignment_1.py:35]: Data not in the project.

**Review by dafrenchyman — APPROVED** [review]
First off, I gave you some extra points for the nice bash script. I basically wrote the exact same script I was running on everyone’s repo when I would pull them to fix and install all the packages. By the time I got to grading your work it was nice to see that someone had thought of that too! You did lose some points for the data not being in the project, I had to copy it over. Other than that, everything was great. Code ran without any issues, was easy to read and follow. Great comments, etc. Great job.

**Pull request merged**